src/plugins/BaseObjectSerializer.ts

```diff
@@ -78,7 +78,15 @@ export class BaseObjectSerializer {
   ): Promise<Map<string, any>> {
     const converted = new Map<string, any>()
 
-    for (const key of Object.keys(o)) {
+    const getters = Object.entries(Object.getOwnPropertyDescriptors(Reflect.getPrototypeOf(o)))
+      .filter(([key, descriptor]) => typeof descriptor.get === 'function' && key !== '__proto__')
+      .map(([key]) => key)
```

Comment:
this looks rather convoluted... I'll do some reading on this. Have you checked with Fabians/Andrew if this is the best way to achieve this?

Reply:
I haven't but I will do that now